src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/MarkdownInterpreter.cs

```diff
@@ -38,7 +39,15 @@ private static string MarkupCore(string content, IProcessContext context, string
             (context.FileLinkSources).Merge(mr.FileLinkSources);
             (context.UidLinkSources).Merge(mr.UidLinkSources);
             (context.Dependency).UnionWith(mr.Dependency);
-            return mr.Html;
+
+            HtmlDocument htmlDocument = new HtmlDocument();
+            htmlDocument.LoadHtml(mr.Html);
```

Comment:
would this change introduce extra perf cost?

Reply:
Yes, it could be. We can use pure string operation with simple check `if (mr.Html.StartWith("<p"))...`.